include/deal.II/numerics/solution_transfer.h

```diff
@@ -138,332 +101,241 @@ DEAL_II_NAMESPACE_OPEN
 
   /**
    * Same as the previous function. It interpolates only one function. It
    * assumes the vectors having the right sizes (i.e.
    * <tt>in.size()==n_dofs_old</tt>, <tt>out.size()==n_dofs_refined</tt>)
    *
-   * Multiple calling of this function is NOT allowed. Interpolating several
-   * functions can be performed in one step by using <tt>interpolate (all_in,
-   * all_out)</tt>
+   * Multiple calling of this function is NOT allowed. Interpolating
+   * several functions can be performed in one step by using
+   * <tt>interpolate (all_in, all_out)</tt>
    */
   void
-  interpolate(const VectorType &in, VectorType &out) const;
+  interpolate(VectorType &out);
 
   /**
-   * Determine an estimate for the memory consumption (in bytes) of this
-   * object.
+   * Prepare the serialization of the given vector. The serialization is
+   * done by Triangulation::save(). The given vector needs all information
+   * on the locally active DoFs (it must be ghosted). See documentation of
+   * this class for more information.
    */
-  std::size_t
-  memory_consumption() const;
+  void
+  prepare_for_serialization(const VectorType &in);
+
+  /**
+   * Same as the function above, only for a list of vectors.
+   */
+  void
+  prepare_for_serialization(const std::vector<const VectorType *> &all_in);
 
   /**
-   * Exception
+   * Execute the deserialization of the given vector. This needs to be
+   * done after calling Triangulation::load(). The given vector must be a
+   * fully distributed vector without ghost elements. See documentation of
+   * this class for more information.
    */
-  DeclExceptionMsg(ExcNotPrepared,
-                   "You are attempting an operation for which this object is "
-                   "not prepared. This may be because you either did not call "
-                   "one of the prepare_*() functions at all, or because you "
-                   "called the wrong one for the operation you are currently "
-                   "attempting.");
+  void
+  deserialize(VectorType &in);
+
 
   /**
-   * Exception
+   * Same as the function above, only for a list of vectors.
    */
-  DeclExceptionMsg(
-    ExcAlreadyPrepForRef,
-    "You are attempting to call one of the prepare_*() functions "
-    "of this object to prepare it for an operation for which it "
-    "is already prepared. Specifically, the object was "
-    "previously prepared for pure refinement.");
+  void
+  deserialize(std::vector<VectorType *> &all_in);
 
   /**
-   * Exception
+   * Reinit this class to the state that it has directly after calling the
+   * constructor.
    */
-  DeclExceptionMsg(
-    ExcAlreadyPrepForCoarseAndRef,
-    "You are attempting to call one of the prepare_*() functions "
-    "of this object to prepare it for an operation for which it "
-    "is already prepared. Specifically, the object was "
-    "previously prepared for both coarsening and refinement.");
+  void
+  clear();
```

Comment:
This function does nothing. Do you want to keep it for compatibility purposes? If so, please deprecate it.